integrations/octopus/main.py

```diff
@@ -0,0 +1,114 @@
+from enum import StrEnum
+from typing import Any, Dict
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from client import OctopusClient
+
+
+class ObjectKind(StrEnum):
+    PROJECT = "project"
+    DEPLOYMENT = "deployment"
+    RELEASE = "release"
+    TARGET = "target"
+
+
+async def init_client() -> OctopusClient:
+    client = OctopusClient(
+        ocean.integration_config["octopus_api_key"],
+        ocean.integration_config["octopus_url"],
+    )
+    return client
+
+
+async def setup_application() -> None:
+    app_host = ocean.integration_config.get("app_host")
+    if not app_host:
+        logger.warning(
+            "App host was not provided, skipping webhook creation. "
+            "Without setting up the webhook, you will have to manually initiate resync to get updates from Octopus."
+        )
+        return
+    octopus_client = await init_client()
+    existing_subscriptions = await octopus_client.get_subscriptions()
+    existing_webhook_uris = {
+        subscription.get("EventNotificationSubscription", {}).get("WebhookURI")
+        for subscription in existing_subscriptions
+    }
+    webhook_uri = f"{app_host}/integration/webhook"
+    if webhook_uri in existing_webhook_uris:
+        logger.info(f"Webhook already exists with URI: {webhook_uri}")
+    else:
+        await octopus_client.create_subscription(app_host)
+        logger.info(f"Webhook created with URI: {webhook_uri}")
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    octopus_client = await init_client()
+    async for projects in octopus_client.get_projects():
+        logger.info(f"Received project batch with {len(projects)} projects")
+        yield projects
+

```

Comment:
If you take my comment above in the client.py into consideration, we would have one global resync 

```python
@ocean.on_resync()
async def on_global_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
    octopus_client = await init_client()
    async for resource_batch in octopus_client.get_paginated_resources(kind):
        logger.info(f"Received length  {len(resource_batch)} of {kind} ")
        yield resource_batch
```